src/python/WMCore/MicroService/Unified/MSCore.py

```diff
@@ -0,0 +1,130 @@
+"""
+File       : MSCore.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSCore class provides core functionality of the MS.
+"""
+# futures
+from __future__ import division, print_function
+
+# system modules
+import time
+
+# WMCore modules
+from WMCore.MicroService.Unified.Common import getMSLogger
+from WMCore.Services.ReqMgr.ReqMgr import ReqMgr
+from WMCore.Services.ReqMgrAux.ReqMgrAux import ReqMgrAux
+from WMCore.Services.PhEDEx.PhEDEx import PhEDEx
+
+
+class MSCore(object):
+    """
+    This class provides core functionality for both
+    MSTransferor and MSMonitor classes.
+    """
+
+    def __init__(self, msConfig, logger=None):
+        """
+        Provides setup for MSTransferor and MSMonitor classes
+
+        :param config: MS service configuration
+        :param logger: logger object (optional)
+        """
+        self.logger = getMSLogger(getattr(msConfig, 'verbose', False), logger)
+        self.msConfig = msConfig
+        self.logger.info(
+            "Configuration including default values:\n%s", self.msConfig)
+
+        self.reqmgr2 = ReqMgr(self.msConfig['reqmgrUrl'], logger=self.logger)
+        self.reqmgrAux = ReqMgrAux(self.msConfig['reqmgrUrl'],
+                                   httpDict={'cacheduration': 60},
+                                   logger=self.logger)
+
+        # eventually will change it to Rucio
+        self.phedex = PhEDEx(httpDict={'cacheduration': 10 * 60},
+                             dbsUrl=self.msConfig['dbsUrl'], logger=self.logger)
+
+    def unifiedConfig(self):
+        """
+        Fetches the unified configuration
+        :return: unified configuration content
+        """
+        return self.reqmgrAux.getUnifiedConfig(docName="config")
```

Comment:
```suggestion
        res = self.reqmgrAux.getUnifiedConfig(docName="config")
        if res:
            return res[0]
        else:
            return {}
```

Reply:
I fixed it in more conservative way, I added check to res to be list.

In general, return types of WMCore APIs is somewhat fragile and inconsistent, but it is a different story.